javascript/modules/client/src/internal/interfaces/client.ts

```diff
@@ -315,3 +338,24 @@ export type SubgraphBalance = {
   balance: string;
   lastUpdated: string;
 };
+
+export enum SubgraphTransferType {
+  DEPOSIT = "Deposit",
+  WITHDRAW = "Withdraw",
+}
+export type SubgraphTransferListItem = {
+  amount: string;
+  createdAt: string;
+  reference: string;
+  transaction: string;
+  type: SubgraphTransferType;
+  to: string;
+  sender: string;
+  token: SubgraphErc20Token;
+};
+export type SubgraphErc20Token = {
+  id: string;
+  name: string;
+  symbol: string;
+  decimals: string;
+};
```

Comment:
Couldn't this be recycled from existing types? `id`=>`address`?

Reply:
Yes in this case id => address, its how its specified in subgraph.
The most similar one is `Erc20TokenDetails` that has an address instead of id, so what can make `Erc20TokenBase` with the 3 common properties and extend it